2009-11-29 Felix Zielcke <fzielcke@z-51.de>

	* util/misc.c (make_system_path_relative_to_its_root): Correctly cope with
	mount points.
=== modified file 'util/misc.c'
--- util/misc.c	2009-11-25 23:10:02 +0000
+++ util/misc.c	2009-11-29 19:19:28 +0000
@@ -500,7 +500,17 @@ make_system_path_relative_to_its_root (c
 
       /* buf is another filesystem; we found it.  */
       if (st.st_dev != num)
-	break;
+	{
+	  /* offset == 0 means path given is the mount point.  */
+	  if (offset == 0)
+	    {
+	      free (buf);
+	      free (buf2);
+	      return strdup ("/");
+	    }
+	  else
+	    break;
+	}
 
       offset = p - buf;
       /* offset == 1 means root directory.  */

